server/src/main/java/org/apache/druid/curator/CuratorModule.java

```diff
@@ -79,10 +80,24 @@ public CuratorFramework makeCurator(CuratorConfig config, EnsembleProvider ensem
           StringUtils.format("%s:%s", config.getZkUser(), config.getZkPwd()).getBytes(StandardCharsets.UTF_8)
       );
     }
+
+    final Function<Void, Void> exitFunction = new Function<Void, Void>()
+    {
+      @Override
+      public Void apply(Void someVoid)
+      {
+        log.error("Zookeeper can't be reached, forcefully stopping lifecycle...");
+        lifecycle.stop();
+        log.error("Zookeeper can't be reached, forcefully stopping virtual machine...");
+        System.exit(1);
+        return null;
+      }
+    };
+
     final CuratorFramework framework = builder
         .ensembleProvider(ensembleProvider)
         .sessionTimeoutMs(config.getZkSessionTimeoutMs())
-        .retryPolicy(new BoundedExponentialBackoffRetry(BASE_SLEEP_TIME_MS, MAX_SLEEP_TIME_MS, MAX_RETRIES))
+        .retryPolicy(config.getQuitOnConnectFail() ? new BoundedExponentialBackoffRetryWithQuit(exitFunction, BASE_SLEEP_TIME_MS, MAX_SLEEP_TIME_MS, MAX_RETRIES) : new BoundedExponentialBackoffRetry(BASE_SLEEP_TIME_MS, MAX_SLEEP_TIME_MS, MAX_RETRIES))
```

Comment:
Line longer than 120 cols. Extract a variable.

Reply:
This has been extracted to a variable, but I couldn't find a clean way to do the conditional on multiple lines, so I just did an if-else refactor.